tap_sailthru/streams.py

```diff
@@ -467,38 +405,85 @@ def prepare_request_payload(
         Returns:
             A dictionary containing the request payload.
         """
-        return {'primary': 1}
+        return {"primary": 1}
 
     def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
-        return {
-            "list_id": record["list_id"],
-            "list_name": record["name"]
-        }
+        return {"list_id": record["list_id"], "list_name": record["name"]}
 
     def parse_response(
-        self,
-        response: SailthruResponse,
-        context: Optional[dict]
+        self, response: SailthruResponse, context: Optional[dict]
     ) -> Iterable[dict]:
         """Parse the response and return an iterator of result rows."""
-        for row in response['lists']:
-            row['account_name'] = self.config.get('account_name')
+        for row in response["lists"]:
+            row["account_name"] = self.config.get("account_name")
             yield row
 
 
+class ListMembersParentStream(sailthruStream):
+    """Manufactured Stream to help with load balancing for List Members stream"""
+
+    name = "list_members_parent"
+    schema_filepath = SCHEMAS_DIR / "lists.json"
+
+    def request_records(self, context: Optional[dict]) -> Iterable[dict]:
+        lists_query = (
+            """
+            SELECT
+                id as list_id,
+                list_name,
+                max(valid_email_count) email_count
+            FROM `"""
+            + self.config.get("table_id")
+            + """`
+            where
+                account = '"""
+            + self.config.get("account_name")
+            + """'
+                and is_primary
+            group by 1, 2
+            order by 2
+        """
+        )
+        self.logger.info(f"Executing query: {lists_query}")
+        client = bigquery.Client(project="nymag-analytics-157315")
+        results = client.query(lists_query).result()
+
+        list_names = [[] for _ in range(self.config.get("num_chunks"))]
+        list_ids = [[] for _ in range(self.config.get("num_chunks"))]
+        totals = [(0, i) for i in range(self.config.get("num_chunks"))]
+        heapq.heapify(totals)  # create a heap queue
```

Comment:
for someone who has no idea what a heap or a queue is, what does this mean? i know what the are and still can't figure out what a `heap queue` is lol

Reply:
I think what's missing from the description here is something between the line comments and the class docs that's at this level of detail - "Queries BQ to get all lists, and then distributes them into ..."